### DISCUSSION TITLE
Why is `jsc.baseUrl` not resolving the require path?

### DISCUSSION THREAD

**Discussion by devjiwonchoi** [swc-project/swc#8665] [RFC]
Hello, fellow engineers and appreciatable SWC maintainers,

I wish to understand if the following behavior is intended and if so, the reason for it:

### TL;DR

Why is `jsc.baseUrl` not resolving the require path as either relative or absolute?

### Context

Using `transform` or `transformSync` from `@swc/core`.

Based on this error message:

```sh
Error: failed to handle: base_dir(`.`) must be absolute. Please ensure that `jsc.baseUrl` is specified correctly.
```

I've understood that the `baseUrl` path must be absolute which differs from `tsconfig.json`, so I'm passing `process.cwd()`.

### Code

```ts
import { foo } from "src/foo"
console.log(foo)
```

```ts
await transform(code, { jsx: { baseUrl: process.cwd() } })
```

Expected:

```js
import { foo } from "./src/foo" // or a resolved absolute path?
console.log(foo)
```

Output:

```ts
import { foo } from "src/foo"
console.log(foo)
```

**Comment by kdy1** [comment]
If you invoke `transform` with relative `jsc.baseUrl`, which directory it should be resolved from? Current working directory? Input file? There's no clear default, so we require absolute url.

**Reply by devjiwonchoi** [reply]
Yes true! I understand why baseUrl needs to be an absolute path since SWC is compile-focused, and I think the error log is very clear and reasonable.

Since using `jsc.baseUrl` and `jsc.paths` do resolve the import path from `@/foo` to `./src/foo`, why not `src/foo` to `./src/foo` on the same baseUrl?

**Reply by devjiwonchoi** [reply]
I can guess that for `paths` it gives keys to resolve (e.g. `@`) but when only `baseUrl`, SWC needs to validate whether `src/foo` is from node_modules or is on the baseUrl, which is an extra step non-related to compilation... right?

By the way, this is 100% what I was looking for: #7302

**Reply by kdy1** [reply]
I think it's a bug, and swc should resolve it. But IIRC, there's a  wrong check for `jsc.paths` that prevents swc from resolving it

**Reply by devjiwonchoi** [reply]
Sounds great that SWC can support it, let me know anything I can assist you on behalf of solving this issue!
